app/channel.go

```diff
@@ -596,10 +596,40 @@ func (a *App) postChannelPrivacyMessage(user *model.User, channel *model.Channel
 	return nil
 }
 
-func (a *App) RestoreChannel(channel *model.Channel) (*model.Channel, *model.AppError) {
+func (a *App) RestoreChannel(channel *model.Channel, userId string) (*model.Channel, *model.AppError) {
+	if channel.DeleteAt == 0 {
+		return nil, model.NewAppError("undeleteChannel", "api.channel.undelete_channel.undeleted.app_error", nil, "", http.StatusBadRequest)
```

Comment:
I feel like `RESTORE*/restore*` sounds better and consistent than `UNDELETE*/undelete*`.

Reply:
we are using 'Unarchive' with the User interface. 'Undelete' on the client applications. and 'Restore' on the API/server level.   I changed it to Restore